### ISSUE TITLE
Create a small façade to break servo_config dependency from style

### ISSUE THREAD

**Issue by Manishearth** [servo/servo#15460]
`servo_config` is a crate in components/config that pulls in a lot of random stuff.

In gecko mode, most of this is unnecessary. However, we pull this in for Stylo because the style crate needs to check if certain options are set.

Stylo can't set these options anyway. We should create a façade module in style that exposes servo_config things in Servo mode, and exposes dummy functions that just return a constant boolean in Gecko mode.

It would look something like

```rust
// in lib.rs
#[cfg(feature="servo")]
extern crate servo_config;

mod servo_config_facade;


// in servo_config_facade.rs
#[cfg(feature = "servo")]
fn viewport_enabled() -> bool {
 use servo_config::prefs::PREFS;
 PREFS.get("layout.viewport.enabled").as_boolean().unwrap_or(false)
}

#[cfg(feature = "gecko")]
fn viewport_enabled() -> bool {true}

// similar functions for each use of servo_config in the style crate


// in stylesheets.rs
// https://github.com/servo/servo/blob/6d1d4b57838d930e6ce39be720ef6c6dbb74af26/components/style/stylesheets.rs#L972
if servo_config_facade::viewport_enabled() {
 // do viewport thingies
}
```

When done, `./mach build -p style` and `./mach build-geckolib` should both work

**Comment by KiChjang** [comment]
Shouldn't this really be called an interface rather than a façade?

**Comment by Manishearth** [comment]
Meh. Either way.

**Comment by karan1276** [comment]
can i work on this?

**Comment by Manishearth** [comment]
Go ahead!

**Comment by nox** [comment]
I'm -2 on calling this facade. Just make a `style::config` module.

**Comment by karan1276** [comment]
I can't find the `viewport_enabled()` function. Where are these functions located?

**Comment by Manishearth** [comment]
The point of this bug is to define these functions. Right now the style code directly calls into `servo_config` code; it should be calling these functions instead.

**Comment by karan1276** [comment]
Okay, so ill find out where style code calls servo_config variables and replace with these functions. Great ill do that

**Comment by karan1276** [comment]
So i have written some code and i want to test it. I am running a VM with just 1 core and only 2 GB RAM (yes not the best machine). Building entire servo project takes a lot of time (+1 hr) Is there a faster way to test my code? What if i just want to run/compile some parts?

**Comment by Manishearth** [comment]
In this case it's a refactor so you just need to build it. Servo does take more than an hour to compile the first time around, but is faster (minutes) for most changes after that)

Is there a reason why you need a VM? Servo compiles on Windows.

**Comment by jdm** [comment]
Have you made progress on this, @karan1276? I know you've been working on another task.

**Comment by karan1276** [comment]
@Manishearth Sorry for late reply, somehow i didn't read that. That issue is resolved now :)

@jdm Yeah i am really done with this, but some force push messed up my branch. Ill just clean that and commit it today

**Comment by mbrubeck** [comment]
Fixed by #16297.
